libraries/chain/controller.cpp

```diff
@@ -666,22 +666,29 @@ struct building_block {
                } else {
                   fork_db.apply_if<void>([&](const auto& forkdb) {
                      auto branch = forkdb.fetch_branch(parent_id());
+                     std::optional<quorum_certificate> qc;
                      for( auto it = branch.begin(); it != branch.end(); ++it ) {
-                        auto qc = (*it)->get_best_qc();
+                        qc = (*it)->get_best_qc();
                         if( qc ) {
                            EOS_ASSERT( qc->block_num <= block_header::num_from_id(parent_id()), block_validate_exception,
                                        "most recent ancestor QC block number (${a}) cannot be greater than parent's block number (${p})",
                                        ("a", qc->block_num)("p", block_header::num_from_id(parent_id())) );
-                           auto qc_claim = qc_claim_t{ qc->block_num, (*it)->timestamp(), qc->qc.is_strong() };
+                           auto claim = qc_claim { qc->block_num, qc->qc.is_strong() };
                            if( bb.parent.is_needed(*qc) ) {
-                              qc_data = qc_data_t{ *qc, qc_claim };
+                              qc_data = qc_data_t{ *qc, claim };
                            } else {
-                              qc_data = qc_data_t{ {},  qc_claim };
+                              qc_data = qc_data_t{ {},  claim };
                            }
                            break;
                         }
                      }
+
+                     if (!qc) {
+                        dlog("IF genesis Block");
+                        qc_data = qc_data_t{ {}, bb.parent.core.latest_qc_claim() };
```

Comment:
Could this not be stored on the IF genesis block so that it can be found above?

Reply:
This happens when current block (bb.parent) is the IF genesis block. It does not have a most recent ancestor which has a QC. We need to construct a default QC claim.